src/core/qgsfeature.cpp

```diff
@@ -313,3 +313,16 @@ QDataStream& operator>>( QDataStream& in, QgsFeature& feature )
   feature.setValid( valid );
   return in;
 }
+
+uint qHash( const QgsFeature& key, uint seed )
```

Comment:
Should be included too. Originally thought of it but when implementing it forgot about it when I was in a rush to leave and quickly wanted this pull request in place.
Good catch!